### ISSUE TITLE
CommandLineUtils is obsolete

### ISSUE THREAD

**Issue by yecril71pl** [dotnet/Scaffolding#1760]
The package [CommandLineUtils](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.commandlineutils) is obsolete and should not be used.

Use the package  [CommandLineUtils](https://natemcmaster.github.io/CommandLineUtils/) instead.

**Comment by tebeco** [comment]
I'm missing something here,
* I'm seeing an announcement in aspnet/Announcments with a linked ticket from 2017 which were already closed at that time
* It says it affect 1.x and 2.0 which both reached EOL times ago

What's the reason of dispatching that announcement on a public repo about a version of AspNet that
* is not maintained
* is not supposed to be used ?


My question here is, if we're subscribing to a repo to get notification on relevant change, should we receive some like that one ?

**Comment by martincostello** [comment]
As noted in the repository's issue template, the aspnet/Announcements repo is intended for announcements by the ASP.NET Core team themselves, not users of GitHub in general.

![image](https://user-images.githubusercontent.com/1439341/146652505-0a180958-cff1-4627-8669-857f639826c2.png)

I'm not sure you meant to broadcast this on your own initiative to the over 4,000 people who subscribe to that repository for announcements:

![image](https://user-images.githubusercontent.com/1439341/146652523-94f5ab60-caed-430a-9b12-ef17f260752e.png)

I haven't put this comment on that issue because I don't think all those people would appreciate also receiving this reply clarifying that as well.

**Comment by tebeco** [comment]
@martincostello, just to be sure, I suspect you're not addressing that comment to me ;) ?

**Comment by martincostello** [comment]
@tebeco Correct, I think we must have been writing a reply at the same time.

**Comment by yecril71pl** [comment]
> I'm not sure you meant to broadcast this on your own initiative to the over 4,000 people who subscribe to that repository for announcements:
> 
> ![image](https://user-images.githubusercontent.com/1439341/146652523-94f5ab60-caed-430a-9b12-ef17f260752e.png)
> 

Of course I did not mean to broadcast.  But that information is important, hard to find and I found no other way of getting it there (where the authors of the Migration documents believe it belongs).

**Comment by martincostello** [comment]
You should have created an issue bringing the missing information to the attention of the product team in the aspnet/Homerepo, and then they would have dealt with it as appropriate.

From the announcement repo's README:

> This repo is for team announcements only. To file a bug or start a discussion, please find the appropriate repo in https://github.com/aspnet and create a new issue. For general issue discussions please go to https://github.com/aspnet/Home/issues and create a new issue.

**Comment by yecril71pl** [comment]
I do not understand what *general issue discussions* are.  It looks like they are supposed to be discussions about existing general issues (and it is still unknown what *general issues* are supposed to be).  I would have done that if you had mentioned *missing announcements* explicitly.

**Comment by martincostello** [comment]
Personally I would take _"General"_ to mean _"where things that don't have a more specific home go"_.

Creating an issue in the Announcements repo when you're not a member of the team still explicitly goes against what's requested in the issue templates and the README, whether or not you know where it _should_ go.

> I would have done that if you had mentioned missing announcements explicitly.

I didn't write the README, nor am I a member of the team that owns it - I just got the announcement like everyone else when you created the issue.

**Comment by yecril71pl** [comment]
> Personally I would take _"General"_ to mean _"where things that don't have a more specific home go"_.

There is [no category](https://github.com/dotnet/aspnetcore/issues/new/choose) for such issues.  The announcement should be labelled 2.0.0, otherwise the migration search will not find it, but I have nowhere the report it.

**Comment by tebeco** [comment]
I'm still not understanding the need to dispacth a 4K user notification
* in place of another team
* a dead lib that was already replaced
* 1.x and 2.0 have reached EOL
* Creating a new project today using supported TFM (3.1.x / 5.0.x and 6.x) will not pull down transitivly this package

So that does not seems like an "Announcement", I'm not sure I'd like to get pinged about something that changed years ago / should not affect any "supported TFM" today

On the other hand you pointed out that this current repo does **explicitly** depends on that package and maybe that's the actual issue.
The current tooling from this repo might need tweaking to spot old dependency that did not received update since a certain amount of time ?
I'm not sure if that would qualify as a ping to **4k users listening to another team/repo announcment**

**Comment by yecril71pl** [comment]
> On the other hand you pointed out that this current repo does **explicitly** depends on that package and maybe that's the actual issue. The current tooling from this repo might need tweaking to spot old dependency that did not received update since a certain amount of time ? I'm not sure if that would qualify as a ping to **4k users listening to another team/repo announcment**

Sure, I wanted to fill in missing info, I did not want to ping the whole world about it.

**Comment by martincostello** [comment]
If you read the README for the announcements repo it points you to `aspnet/Home`, which is where you should have created an issue. The category link you just pointed to is in `dotnet/aspnet`, a completely different repo.

**Comment by tebeco** [comment]
> Sure, I wanted to fill in missing info, I did not want to ping the whole world about it.

then why not just create this issue here and that's all ?

**Comment by yecril71pl** [comment]
> > Sure, I wanted to fill in missing info, I did not want to ping the whole world about it.
> 
> then why not just create this issue here and that's all ?

I cannot say something is obsolete when it is not documented it is obsolete.

**Comment by tebeco** [comment]
sooo you declare it obsolete yourself on another repo, in the name of another team, ping 4k users, and re-use that here ?

Also we could continue to discuss for hours here, that would not be constructive.

Yes you could just have created this one, and you could have raised it as a question
"it look like it's obsolete and we might want to migrate blablabla, what can we do about it ? here I provided a PR to help in case you wanted it ... blablabla ...." same meaning, less direct if you're not sure and opening the discsussion.

That said, I wish you all happy holidays

**Comment by martincostello** [comment]
Well that's the point we're trying to get at - you should have created an issue somewhere other than the announcement repo, and then the product team would have taken appropriate action. Instead you ignored the instructions in the announcements repo, and consequently told over 4,000 people about it.

Even if that's not what you intended to do and it's a genuine mistake (we all make mistakes), that is what happened, and why the instructions were put in the repo and the repo's issue templates - to try and stop this from happening in the first place.

**Comment by yecril71pl** [comment]
> If you read the README for the announcements repo it points you to `aspnet/Home`, which is where you should have created an issue. The category link you just pointed to is in `dotnet/aspnet`, a completely different repo.

What I read is the template at [Use aspnet/AspNetCore instead](https://github.com/aspnet/Announcements/issues/new?assignees=&labels=&template=Default.md).  To stop this from happening, you need to update the template.

**Comment by martincostello** [comment]
> you need to update the template.

I'm not a member the team - stop telling **me** to do things.

Also, the link you linked to makes it fairly clear to me that you still shouldn't have created it in the first place from the five repetitions of `!!! DO NOT POST ISSUES IN THIS REPO !!!`.

![image](https://user-images.githubusercontent.com/1439341/146653860-68e0bdd3-72f3-46db-b67e-b9a553ea0934.png)

Even if the link in that template is out of date (because `aspnet/AspNetCore` got moved to `dotnet/aspnetcore`) and it doesn't have a category you think appropriate in its issue templates, it doesn't seem to justify the action of ignoring all the requests to _not_ open the issue in the first place (there's three attempts before you even get as far as getting to the issue template; the README and the text of the issue template summaries highlighted in red in [my original reply to this issue](https://github.com/dotnet/Scaffolding/issues/1760#issuecomment-997268092)).
